mono/metadata/mono-perfcounters.c

```diff
@@ -1304,52 +1307,52 @@ custom_get_impl (SharedCategory *cat, MonoString *counter, MonoString* instance,
 }
 
 static const CategoryDesc*
-find_category (MonoString *category)
+find_category (const gunichar2* category, int category_length)
 {
 	int i;
+	const CategoryDesc* desc = predef_categories;
 	for (i = 0; i < NUM_CATEGORIES; ++i) {
-		if (mono_string_compare_ascii (category, predef_categories [i].name) == 0)
-			return &predef_categories [i];
+		if (perfcounter_utf16_equal_ascii (category, category_length, desc->name, desc->name_length))
+			return desc;
+		++desc;
 	}
 	return NULL;
 }
 
 void*
-mono_perfcounter_get_impl (MonoString* category, MonoString* counter, MonoString* instance,
-		MonoString* machine, int *type, MonoBoolean *custom)
+ves_icall_System_Diagnostics_PerformanceCounter_GetImpl (
+	const gunichar2* category, int category_length,
+	const gunichar2* counter, int counter_length,
+	const gunichar2* instance, int instance_length,
+	int *type, MonoBoolean *custom, MonoError *error)
+// previously mono_perfcounter_get_impl
```

Comment:
Why not remove that?

Reply:
It is much easier to sere here than in history and it has been useful looking at other branches at the same time.